lib/state/ui/reducer.ts

```diff
@@ -90,9 +90,20 @@ const dialogs: A.Reducer<T.DialogType[]> = (state = [], action) => {
   switch (action.type) {
     case 'CLOSE_DIALOG':
       return state.slice(0, -1);
-
+    case 'TRASH_TAG':
+      return state.filter(
+        (element) => element.type !== 'TRASH-TAG-CONFIRMATION'
+      );
+    case 'REMOTE_TAG_DELETE':
+      return state.filter(
+        (element) =>
+          element.tagName !== tagHashOf(action.tagHash) ||
```

Comment:
I'm a bit lost in the negative-or logic with the filter. would it be clearer if we stated the specific dialog we want to get rid of instead of the qualifications for which ones can stay?

```ts
return state.filter(element => !(
	element.type === 'TRASH-TAG-CONFIRMATION' &&
	element.tagName === tagHashOf(action.tagHash)
))
```

Reply:
Thank you! I was confusing myself with this logic as well. This suggestion is so much more readable.